Mathlib/RingTheory/PowerSeries/Basic.lean

```diff
@@ -502,6 +527,37 @@ theorem X_dvd_iff {φ : R⟦X⟧} : (X : R⟦X⟧) ∣ φ ↔ constantCoeff R φ
   · intro m hm
     rwa [Nat.eq_zero_of_le_zero (Nat.le_of_succ_le_succ hm)]
 
+instance [NoZeroDivisors R] : NoZeroDivisors R⟦X⟧ where
+  eq_zero_or_eq_zero_of_mul_eq_zero {φ ψ} h := by
+    classical
+    rw [or_iff_not_imp_left]
+    intro H
+    have ex : ∃ m, coeff R m φ ≠ 0 := by
+      contrapose! H
+      exact ext H
+    let m := Nat.find ex
+    have hm₁ : coeff R m φ ≠ 0 := Nat.find_spec ex
+    have hm₂ : ∀ k < m, ¬coeff R k φ ≠ 0 := fun k => Nat.find_min ex
+    simp only [ne_eq, Decidable.not_not] at hm₂
+    ext n
+    rw [(coeff R n).map_zero]
+    induction' n using Nat.strong_induction_on with n ih
+    replace h := congr_arg (coeff R (m + n)) h
+    rw [LinearMap.map_zero, coeff_mul, Finset.sum_eq_single (m, n)] at h
+    · replace h := NoZeroDivisors.eq_zero_or_eq_zero_of_mul_eq_zero h
+      rw [or_iff_not_imp_left] at h
+      exact h hm₁
```

Comment:
```suggestion
    · simp_all
```

Reply:
The simp_all doesn't work here, but it helped simplify the proof.